providers/aws-ec2/src/test/java/org/jclouds/aws/ec2/features/InternetGatewayApiMockTest.java

```diff
@@ -0,0 +1,143 @@
+      enqueueRegions(DEFAULT_REGION);
+      enqueueXml(DEFAULT_REGION, "/detach_internet_gateway.xml");
+
+      final Boolean detached = gatewayApi().detachInternetGateway(DEFAULT_REGION, "igw-fada7c9c", "vpc-6250b91b");
+      assertTrue(detached);
+
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=DetachInternetGateway&InternetGatewayId=igw-fada7c9c&VpcId=vpc-6250b91b");
+
+   }
+
+   public void getInternetGateway() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueueXml(DEFAULT_REGION, "/get_internet_gateway.xml");
+
+      final String igwId = "igw-fada7c9c";
+      final FluentIterable<InternetGateway> internetGateways =
+         gatewayApi().describeInternetGatewaysInRegion(DEFAULT_REGION, igwId);
+
+      final ImmutableList<InternetGateway> gateways = internetGateways.toList();
+      assertEquals(gateways.size(), 1);
+      assertEquals(gateways.get(0).id(), igwId);
+      assertEquals(gateways.get(0).tags().get("Name"), "get_internet_gateway_test");
+
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=DescribeInternetGateways&InternetGatewayId.1=igw-fada7c9c");
+
+   }
+
+   public void describeInternetGateways() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueueXml(DEFAULT_REGION, "/describe_internet_gateways.xml");
+
+      final FluentIterable<InternetGateway> internetGateways =
+         gatewayApi().describeInternetGatewaysInRegion(DEFAULT_REGION);
+
+      final List<InternetGateway> gateways = Lists.newArrayList(internetGateways.toList());
+
+      assertEquals(gateways.size(), 3);
+      final ImmutableMap<String, InternetGateway> asMap =
+         ImmutableMap.of(gateways.get(0).id(), gateways.get(0),
+            gateways.get(1).id(), gateways.get(1),
+            gateways.get(2).id(), gateways.get(2));
+
+      assertEquals(asMap.get("igw-fada7c9c").tags().get("Name"), "describe_internet_gateways_test");
+      final InternetGatewayAttachment gw6bca130c = asMap.get("igw-6bca130c").attachmentSet().iterator().next();
+      assertEquals(gw6bca130c.vpcId(), "vpc-a13d29c6");
+      assertEquals(gw6bca130c.state(), InternetGatewayAttachment.State.AVAILABLE);
+
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=DescribeInternetGateways");
+
+   }
+
+   public void deleteInternetGateway() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueueXml(DEFAULT_REGION, "/delete_internet_gateway.xml");
+
+      final boolean deleted = gatewayApi().deleteInternetGateway(DEFAULT_REGION, "igw-fada7c9c");
+      assertTrue(deleted);
+
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=DeleteInternetGateway&InternetGatewayId=igw-fada7c9c");
+   }
+
+   private InternetGatewayApi gatewayApi() {
+      return api().getInternetGatewayApi().get();
+   }
+}
```

Comment:
Add the tests that exercise the fallbacks defined in the API methods by enqueuing 404 responses.